keras_core/export/export_lib.py

```diff
@@ -0,0 +1,574 @@
+        tf.saved_model.save(
+            self, filepath, options=options, signatures=signatures
+        )
+        # Print out available endpoints
+        endpoints = "\n\n".join(
+            _print_signature(getattr(self, name), name)
+            for name in self._endpoint_names
+        )
+        io_utils.print_msg(
+            f"Saved artifact at '{filepath}'. "
+            "The following endpoints are available:\n\n"
+            f"{endpoints}"
+        )
+
+    def _get_concrete_fn(self, endpoint):
+        """Workaround for some SavedModel quirks."""
+        if endpoint in self._endpoint_signatures:
+            return getattr(self, endpoint)
+        else:
+            traces = getattr(self, endpoint)._trackable_children("saved_model")
+            return list(traces.values())[0]
+
+    def _get_variables_used_by_endpoints(self):
+        fns = [self._get_concrete_fn(name) for name in self._endpoint_names]
+        return _list_variables_used_by_fns(fns)
+
+    def _filter_and_track_resources(self):
+        """Track resources used by endpoints / referenced in `track()` calls."""
+        # Start by extracting variables from endpoints.
+        fns = [self._get_concrete_fn(name) for name in self._endpoint_names]
+        tvs, ntvs = _list_variables_used_by_fns(fns)
+        self._all_variables = list(tvs + ntvs)
+
+        # Next, track lookup tables.
+        # Hopefully, one day this will be automated at the tf.function level.
+        self._misc_assets = []
+        from keras_core.layers import IntegerLookup
+        from keras_core.layers import StringLookup
+        from keras_core.layers import TextVectorization
+
+        if hasattr(self, "_tracked"):
+            for root in self._tracked:
+                descendants = tf.train.TrackableView(root).descendants()
+                for trackable in descendants:
+                    if isinstance(
+                        trackable,
+                        (IntegerLookup, StringLookup, TextVectorization),
+                    ):
+                        self._misc_assets.append(trackable)
+
+
+def export_model(model, filepath):
+    export_archive = ExportArchive()
+    export_archive.track(model)
+    if isinstance(model, (Functional, Sequential)):
+        input_signature = tf.nest.map_structure(_make_tensor_spec, model.inputs)
+        if isinstance(input_signature, list) and len(input_signature) > 1:
+            input_signature = [input_signature]
+        export_archive.add_endpoint("serve", model.__call__, input_signature)
+    else:
+        save_spec = model._get_save_spec()
+        if not save_spec:
+            raise ValueError(
+                "The model provided has never called. "
+                "It must be called at least once before export."
+            )
+        input_signature = [save_spec]
+        export_archive.add_endpoint("serve", model.__call__, input_signature)
+    export_archive.write_out(filepath)
+
+
+class ReloadedLayer(Layer):
```

Comment:
Upon reflection, I think this is a pretty unfortunate name. I wonder if we should rename it to `TFSMLayer` or something to that extent? In the future we could have other formats as well, like `ONNXLayer`. It would make sense since it can work with more than what's saved with ExportArchive -- any SavedModel will do.

To note, this object doesn't seem to be exported to the public API yet.

Reply:
Good idea, I've changed it to TFSMLayer, thanks.